### PR TITLE
1608130: Document how to update android-components

### PR DESCRIPTION
**Pull request by mdboom**

### CODE REVIEW

File: docs/dev/cut-a-new-release.md [hunk 1]
```diff
@@ -204,3 +205,23 @@ If you need to release a hotfix for a previously released version (that is: not
     * This is important so that no changes are lost.
     * This might have merge conflicts with the `master` branch, which you need to fix before it is merged.
 8. Once the above pull request lands, delete the support branch.
+
+## Upgrading android-components to a new version of Glean
+
+On Android, many users of Glean import it through its wrapper in [`android-components`](https://github.com/mozilla-mobile/android-components).
```
Comment (Dexterp37) [inline docs/dev/cut-a-new-release.md]:
```suggestion
On Android, Mozilla products consume the Glean SDK through its wrapper in [`android-components`](https://github.com/mozilla-mobile/android-components).
```

Comment (Dexterp37) [reply docs/dev/cut-a-new-release.md]: This is actually a requirement for Mozilla products :-)

File: docs/dev/cut-a-new-release.md [hunk 2]
```diff
@@ -204,3 +205,23 @@ If you need to release a hotfix for a previously released version (that is: not
...
+Therefore, when a new Glean release is made, `android-components` must also be updated.
```
Comment (Dexterp37) [inline docs/dev/cut-a-new-release.md]:
```suggestion
Therefore, when a new Glean SDK release is made, `android-components` must also be updated.
```

File: docs/dev/cut-a-new-release.md [hunk 3]
```diff
@@ -204,3 +205,23 @@ If you need to release a hotfix for a previously released version (that is: not
...
+
+After following one of the above instructions to make a Glean release:
```
Comment (Dexterp37) [inline docs/dev/cut-a-new-release.md]:
```suggestion
After following one of the above instructions to make a Glean SDK release:
```

File: docs/dev/cut-a-new-release.md [hunk 4]
```diff
@@ -204,3 +205,23 @@ If you need to release a hotfix for a previously released version (that is: not
...
+
+1) Ensure that CI has completed and the artifacts have published to [the Glean GitHub releases page](https://github.com/mozilla/glean/releases/).
+
+2) Publish Glean to [Mozilla's Maven repository](https://maven.mozilla.org/). 
+   This is currently a manual step that can only performed by individuals with the necessary credentials.
+   Ask in `#releaseduty-mobile` on Mozilla's internal Slack for assistance.
+   
+3) Create a pull request against `android-components` to update the Glean version with the following changes:
+   
+   - The Glean version is updated in the `mozilla_glean` variable in the `buildSrc/src/main/java/Dependencies.kt` file.
```
Comment (Dexterp37) [inline docs/dev/cut-a-new-release.md]:
```suggestion
   - The Glean version is updated in the `mozilla_glean` variable in the [`buildSrc/src/main/java/Dependencies.kt`](https://github.com/mozilla-mobile/android-components/blob/69546999739ab19d21425e9a98e107e438a3f905/buildSrc/src/main/java/Dependencies.kt#L32) file.
```

File: docs/dev/cut-a-new-release.md [hunk 5]
```diff
@@ -204,3 +205,23 @@ If you need to release a hotfix for a previously released version (that is: not
...
+   
+   - The relevant parts of the Glean changelog copied into the `android-components` changelog.
```
Comment (Dexterp37) [inline docs/dev/cut-a-new-release.md]:
```suggestion
   - The relevant parts of the Glean changelog copied into the [`android-components` changelog](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md), in the top section with the`-SNAPSHOT` suffix in the title.
```

Comment (mdboom) [reply docs/dev/cut-a-new-release.md]: That's only relevant if they are releasing snapshots, which IIUC is not the long term plan.  I'd be fine with just saying "top section" here, though.

Comment (Dexterp37) [reply docs/dev/cut-a-new-release.md]: Sounds good to me!

File: docs/dev/cut-a-new-release.md [hunk 6]
```diff
@@ -204,3 +205,23 @@ If you need to release a hotfix for a previously released version (that is: not
...
 8. Once the above pull request lands, delete the support branch.
+
+## Upgrading android-components to a new version of Glean
+
+On Android, Mozilla products consume the Glean SDK through its wrapper in [`android-components`](https://github.com/mozilla-mobile/android-components).
+Therefore, when a new Glean SDK release is made, `android-components` must also be updated.
+
+After following one of the above instructions to make a Glean SDK release:
+
+1) Ensure that CI has completed and the artifacts have published to [the Glean GitHub releases page](https://github.com/mozilla/glean/releases/).
```
Comment (badboy) [inline docs/dev/cut-a-new-release.md]: While this seems to render as a list just fine, should we use the name `<number>.` format as for other lists in this document, for consistency?

**Pull request merged**